SC_redesign/Crop_and_Soil/soil/phosphorus_cycling/phosphorus_mineralization.py

```diff
@@ -0,0 +1,341 @@
+
+            layer.active_inorganic_phosphorus_content -= phosphorus_mineralized
+            layer.labile_inorganic_phosphorus_content += phosphorus_mineralized
+
+            layer.previous_phosphorus_balance = balance
+
+            stable_to_active_mineralization_amount = self._determine_stable_to_active_phosphorus_mineralization(
+                layer.stable_inorganic_phosphorus_content, layer.active_inorganic_phosphorus_content)
+            layer.stable_inorganic_phosphorus_content -= stable_to_active_mineralization_amount
+            layer.active_inorganic_phosphorus_content += stable_to_active_mineralization_amount
+
+    # --- Static methods ---
+    @staticmethod
+    def _recompute_mean_phosphorus_sorption_parameter(mean_sorption_parameter: float,
+                                                      current_sorption_parameter: float) -> float:
+        """Recalculates the mean sorption parameter based on current day's condition.
+
+        Parameters
+        ----------
+        mean_sorption_parameter : float
+            The mean phosphorus sorption parameter of the given soil layer (unitless)
+        current_sorption_parameter : float
+            The phosphorus sorption parameter of the given soil layer calculated with the layer's current conditions
+                                                                                                            (unitless)
+
+        Returns
+        -------
+        float
+            The mean phosphorus sorption parameter that has been adjusted for the current day's amount of labile
+            inorganic phosphorus present (unitless)
+
+        References
+        ----------
+        SurPhos Fortran code, pminrl.f, lines 48 - 51
+
+        Notes
+        -----
+        The phosphorus sorption parameter, in the words of Pete Vadas, "represents sort of a long term chemical
+        characteristic of the soil and should NOT be calculated every day. There can be big changes in labile P when P
+        is added to soils in fertilizer and manure, and we don’t want PSP changing rapidly." Be recalculating the
+        phosphorus sorption parameter with this equation every day (as opposed to recalculating it with
+        calculate_phosphorus_sorption_parameter() in LayerData), we keep changes in it limited.
+
+        """
+        new_mean_phosphorus_sorption_parameter = ((29 * mean_sorption_parameter) + current_sorption_parameter) / 30
+        return max(0.05, min(0.7, new_mean_phosphorus_sorption_parameter))
+
+    @staticmethod
+    def _determine_phosphorus_imbalance(labile_phosphorus: float, active_phosphorus: float,
+                                        sorption_parameter: float) -> float:
+        """Calculates the imbalance of phosphorus between the labile and active inorganic pools.
+
+        Parameters
+        ----------
+        labile_phosphorus : float
+            Labile inorganic phosphorus content of this soil layer (kg phosphorus / ha)
```

Comment:
I think (kg/ha) is enough.
```suggestion
            Labile inorganic phosphorus content of this soil layer (kg/ ha)
```